cell_classification/segmentation_data_prep_test.py

```diff
@@ -1,24 +1,77 @@
+from calendar import c
+from genericpath import exists
 import os
 import pytest
 import tempfile
 import numpy as np
+import pandas as pd
+import json
 from tifffile import imwrite
 from segmentation_data_prep import SegmentationTFRecords
 
 
 def prep_object():
-    data_prep = SegmentationTFRecords(
-        data_folders="path_to_folders",
-        cell_table_path="path_to_cell_table",
-        conversion_matrix_path="path_to_conversion_matrix",
-        imaging_platform="ImagingPlatform",
-        dataset="Dataset",
-        tile_size=[256, 256],
-        tf_record_path="path_to_tf_record",
-    )
+    with tempfile.TemporaryDirectory() as temp_dir:
+        norm_dict = {"CD8": 1.0, "CD4": 1.0}
+        norm_dict_path = os.path.join(temp_dir, "norm_dict.json")
+        json.dump(norm_dict, open(norm_dict_path, "w"))
+        conversion_matrix = prepare_conversion_matrix()
+        conversion_matrix_path = os.path.join(temp_dir, "conversion_matrix.csv")
+        conversion_matrix.to_csv(conversion_matrix_path, index=False)
+        data_prep = SegmentationTFRecords(
+            data_folders=["list", "of", "data", "folders"],
+            cell_table_path="list_to_cell_table_path",
+            conversion_matrix_path=conversion_matrix_path,
+            imaging_platform="imaging_platform",
+            dataset="dataset",
+            tile_size=[256, 256],
+            tf_record_path=os.path.join(temp_dir, "tf_record_path"),
+            normalization_dict_path=norm_dict_path,
+        )
     return data_prep
 
 
+def prepare_conversion_matrix():
+    col_names = [
+        "cluster_labels",
+        "CD11c",
+        "CD14",
```

Comment:
For testing, it'll make your life much easier if you go with the minimal example. Probably 3-4 markers is enough